### PR TITLE
Badge highlights

### PR DESCRIPTION
**Pull request by qbalukom**

### CODE REVIEW

**Review by Wawrzyn321 — CHANGES_REQUESTED** [review]
Also, would be nice to have some tests just for the logic in our `Badge` wrapper.

File: core-ui/src/components/Extensibility/components/Badge.js [hunk 1]
```diff
@@ -1,16 +1,34 @@
...
+export function Badge({ value, structure, schema, ...props }) {
   const { emptyLeafPlaceholder } = useGetPlaceholder(structure);
+
+  let type = null;
+  if (structure.highlights) {
+    const match = Object.entries(structure.highlights).find(([key, rule]) => {
+      if (Array.isArray(rule)) {
+        return rule.includes(value);
+      } else {
+        return jsonata(rule).evaluate({ value });
```
[the remark this answers is not in the document]
Comment (qbalukom) [reply core-ui/src/components/Extensibility/components/Badge.js:17]: makes sense

File: core-ui/src/components/Extensibility/components/Badge.js [hunk 2]
```diff
@@ -1,16 +1,39 @@
...
+  let type = null;
+  if (structure.highlights) {
+    const match = Object.entries(structure.highlights).find(([key, rule]) => {
+      if (Array.isArray(rule)) {
+        return rule.includes(value);
+      } else {
+        try {
+          return jsonata(rule).evaluate({ value });
+        } catch (e) {
+          console.warn(`invalid rule: ${rule}`);
```
Comment (Wawrzyn321) [inline core-ui/src/components/Extensibility/components/Badge.js:20]:
```suggestion
          console.warn(`invalid rule: ${rule}`, e);
```

File: core-ui/src/components/Extensibility/components/Badge.js [hunk 3]
```diff
@@ -1,16 +1,39 @@
...
   const { emptyLeafPlaceholder } = useGetPlaceholder(structure);
+
+  let type = null;
+  if (structure.highlights) {
+    const match = Object.entries(structure.highlights).find(([key, rule]) => {
+      if (Array.isArray(rule)) {
+        return rule.includes(value);
+      } else {
+        try {
+          return jsonata(rule).evaluate({ value });
```
Comment (Wawrzyn321) [inline core-ui/src/components/Extensibility/components/Badge.js:18]: We use `data` while passing stuff to jsonata, let's be consistent.
```suggestion
          return jsonata(rule).evaluate({ data: value });
```

File: docs/extensibility/display-widgets.md [hunk 4]
```diff
@@ -31,25 +31,32 @@ Text widgets render values as a simple text. This is the default behavior for al
...
 
 ```json
 {
   "path": "status.value",
   "widget": "Badge",
-  "placeholder": "-"
+  "placeholder": "-",
+  "highlights": {
+    "positive": ["yes", "ok"],
+    "negative": "value < 0"
```
Comment (Wawrzyn321) [inline docs/extensibility/display-widgets.md:58]: Same as before
```suggestion
    "negative": "data < 0"
```

**Review by Wawrzyn321 — APPROVED** [review]

File: docs/extensibility/display-widgets.md [hunk 5]
```diff
@@ -31,25 +31,32 @@ Text widgets render values as a simple text. This is the default behavior for al
...
-- rendered as a success: `ready`, `bound`, `running`, `success`, `succeeded`, `ok`.
-- rendered as a warning: `unknown`, `warning`.
-- rendered as an error: `error`, `failure`, `invalid`.
-
-###### Widget-specific parameters
+#### Widget-specific parameters
 
 - **placeholder** - an optional property to change the default empty text placeholder `-` with a custom string.
   If the translation section has a translation entry with the ID that is the same as the **placeholder** string, the translation is used.
+- **highlights** - an optional map of highlight rules. Key refers to type of highlight, while the rule can just be a plain array of values or a string containing a jsonata rule. Allowed keys are `informative` `positive`, `negative` and `critical`.
```
Comment (grego952) [inline docs/extensibility/display-widgets.md:38]:
```suggestion
- **highlights** - an optional map of highlight rules. Key refers to the type of highlight, while the rule can just be a plain array of values or a string containing a jsonata rule. Allowed keys are `informative` `positive`, `negative` and `critical`.
```

**Pull request merged**